library/Opus/Model2/Language.php

```diff
@@ -0,0 +1,395 @@
+    public function setActive($active)
+    {
+        $this->active = $active;
+    }
+
+    /**
+     * Returns reference language name.
+     *
+     * @return string
+     */
+    public function getDisplayName()
+    {
+        return $this->getRefName();
+    }
+
+    /**
+     * Retrieve all Opus\Language instances from the database.
+     *
+     * @return array Array of Opus\Language objects.
+     * @throws ORMException
+     */
+    public static function getAll()
+    {
+        return self::getRepository()->getAll();
+    }
+
+    /**
+     * Get all active languages.
+     *
+     * @return array Array of Opus\Language objects which are active.
+     * @throws ORMException
+     */
+    public static function getAllActive()
+    {
+        return self::getRepository()->getAllActive();
+    }
+
+    /**
+     * Get all active languages.
+     *
+     * @return array Array of Opus\Language objects which are active.
+     * @throws ORMException
+     */
+    public static function getAllActiveTable()
+    {
+        return self::getRepository()->getAllActiveTable();
+    }
+
+    /**
+     * Get properties of language object as array for a specific terminology code
+     *
+     * @param string $code ISO639-2 terminology code to retrieve properties for
+     * @return Language|null Language model or null if object not found in database
+     * @throws ORMException
+     */
+    public static function getLanguageByPart2T($code)
+    {
+        return self::getRepository()->getLanguageByPart2T($code);
+    }
+
+    /**
+     * Returns part2_t language code for locale (part1 code).
+     *
+     * @param string $locale
+     * @return null|string
+     * @throws ORMException
+     */
+    public static function getPart2tForPart1($locale)
+    {
+        return self::getRepository()->getPart2tForPart1($locale);
+    }
+
+    /**
+     * Returns language code for internal language identifier.
+     *
+     * @param string      $language Internal language identifier (e.g. 'deu')
+     * @param null|string $part string Field to use for language code
+     * @return string Language code
+     * @throws ORMException
+     */
+    public static function getLanguageCode($language, $part = null)
+    {
+        return self::getRepository()->getLanguageCode($language, $part);
+    }
+
+    /**
+     * Checks if a language is being used in database.
+     *
+     * @return mixed
+     * @throws ORMException
+     */
+    public function isUsed()
+    {
+        return self::getRepository()->isUsed($this->getPart2T());
+    }
+
+    /**
+     * Returns all languages used in database.
+     *
+     * @return mixed
+     * @throws ORMException
+     */
+    public static function getUsedLanguages()
+    {
+        return self::getRepository()->getUsedLanguages();
+    }
+
+    /**
+     * Removes cached values.
+     *
+     * @throws ORMException
+     */
+    public static function clearCache()
+    {
+        self::getRepository()->clearCache();
```

Comment:
Da wir das Repository nicht mehr cachen, greift das hier vermutlich ins leere. Ist es mit Doctrine überhaupt noch notwendig? Ich denke nicht. Das Caching sollte dem ORM überlassen werden. Diese Funktion ist nur für Tests relevant und für die kann man bei Bedarf Funktionen außerhalb der Model-API zur Verfügung stellen. ich entferne das.

Reply:
Ich hatte vergessen, dass es hier um die UsedLanguages geht. Das Caching kann nicht im LanguageRepository stattfinden und ist sowieso unabhängig von der Datenbank. Das Caching muss in die Modelklasse wandern.

Reply:
Caching ist nun in der Modelklasse.

Reply:
Weil das Caching verschoben wurde, muss auch die Implementation von isUsed angepasst werden.